### DISCUSSION TITLE
Temporary Data Settings

### DISCUSSION THREAD

**Discussion by wood3875** [codefori/vscode-ibmi#1582] [IBM i] [answered]
With two systems configured, one works as expected with ILEDITOR, and the other does not (exact same setting).  

When I connect, I get this:
Code for IBM i will not function correctly until the temporary library has been corrected in the settings.

I have cleared the lib, deleted, restored from another system, cleared again, turned on "Clear Temporary Data Automatically", restarted everything, still errors out. 

I have deleted the system connection and recreated it, still errors out.

Why ?

![image](https://github.com/codefori/vscode-ibmi/assets/146120671/cf21cc04-f8db-4787-bc56-2d7d604dc6e8)
![image](https://github.com/codefori/vscode-ibmi/assets/146120671/86b7a4cb-9d18-4906-9802-59d546561b45)

**Comment by SJLennon** [comment]
@wood3875 
Are you sure that the user has enough authority to the ILEDITOR library in the connection that fails?

**Reply by wood3875** [reply]
100% sure.  I am the owner and creator of both, with *ALL object and data authorities.  Security on both systems are identical and look like this:
                           Object     -----Object------   ------Data-------
User         Group        Authority   O   M   E   A   R   R   A   U   D   E
*PUBLIC                   *USE        X                   X               X
WOODJG                    *ALL        X   X   X   X   X   X   X   X   X   X

**Reply by SJLennon** [reply, marked as answer]
@wood3875  
You might also want to look at the Code for IBM i data under the Output tab.  See if any errors show up there. 

In my connection to PUB400 I see this, because I'm not authorized to the CRTLIB command. You might see something similar, or some other error, for the problem connection.

![image](https://github.com/codefori/vscode-ibmi/assets/67484051/45e98bee-8c31-42da-a9fb-002d2cf29357)

**Reply by wood3875** [reply]
SJLennon - thank you.  I see/know the reason...  We have a MODIFIED version of CRTLIB found in an upper library (library list).   Our modified CRTLIB ensures any/every lib creation gets into the iASP, this is a passive DR target system where the iASP is not on/available, thus the failure.

**I'd LIKE to see a fix whereby, any system command be qualified with QSYS ?  
QSYS/CRTLIB would not present an error that I have currently have no workaround...**


/home/WOODJG: system "CRTLIB LIB(ILEDITOR) TEXT('Code for i temporary objects. May be cleared.')"
{
    "code": 255,
    "signal": null,
    "stdout": "",
    "stderr": "CPFB8ED: Device description DTIIASP not correct for operation."

**Reply by worksofliam** [reply]
@wood3875 

> I'd LIKE to see a fix whereby, any system command be qualified with QSYS ?
QSYS/CRTLIB would not present an error that I have currently have no workaround

I would raise an issue for this. We are also open to PRs if you'd like to add it faster. Thank you!
